tests/unit/changeset-test.js

```diff
@@ -1057,6 +1060,137 @@ module('Unit | Utility | changeset', function(hooks) {
+
+    dummyChangeset.set('profile.firstName', 'Zoe');
+
+    assert.equal(get(dummyChangeset, 'isPristine'), false, 'changeset is not pristine as there is a change');
+
+    assert.equal(mockUserModel.get('profile.firstName'), 'Bob', 'Original model property should stay without changes')
+    assert.equal(mockUserModel.get('profile.lastName'), 'Ross', 'Original model property should stay without changes')
+
+    assert.throws(
+      function() {
+        assert.equal(dummyChangeset.get('pendingData.profile.firstName'), 'Zoe', 'Model belongsTo property should be updated');
+        assert.equal(dummyChangeset.get('pendingData.profile.lastName'), 'Ross', 'Existing property should stay the same');
+      },
+      /Unable to `mergeDeep` with your data. Are you trying to merge two ember-data objects\? Please file an issue with ember-changeset\./,
+      "raised error instance indicates that ember-data models can't be merged"
+    );
+
+    assert.equal(dummyChangeset.get('data.profile.firstName'), 'Bob', 'Original model property should stay without changes')
+    assert.equal(dummyChangeset.get('data.profile.lastName'), 'Ross', 'Original model property should stay without changes')
+  });
+
+
+  todo('#pendingChanges with ember-data model with multiple change steps', async function(assert) {
+    let store = this.owner.lookup('service:store');
+
+    let mockProfileModel = store.createRecord('profile');
+    let mockUserModel = store.createRecord('user', {
+      profile: mockProfileModel,
+      save: function() {
+        return Promise.resolve(this);
+      },
+    });
+
+    let dummyChangeset = Changeset(mockUserModel);
+    assert.equal(get(dummyChangeset, 'isPristine'), true, 'changeset is pristine');
+
+    dummyChangeset.set('profile.firstName', 'Zoe');
+
+    assert.equal(get(mockUserModel, 'profile.firstName'), 'Bob', 'belongsTo property is not modified');
+    assert.equal(get(dummyChangeset, 'profile.firstName'), 'Zoe', 'belongsTo changeset is modified');
+    assert.equal(get(dummyChangeset.pendingData, 'profile.firstName'), 'Zoe', 'pending data should give updated values');
+
+    await dummyChangeset.save();
+
+    assert.equal(get(mockUserModel, 'profile.firstName'), 'Zoe', 'belongsTo property is updated after save');
+    assert.equal(get(dummyChangeset, 'profile.firstName'), 'Zoe', 'belongsTo changeset has new value');
+    assert.equal(get(dummyChangeset.pendingData, 'profile.firstName'), 'Zoe', 'pendingData should have a new value');
+
+    let updatedMockProfileModel = store.createRecord('profile');
+
+    dummyChangeset.set('profile', updatedMockProfileModel);
+
+    assert.equal(get(mockUserModel, 'profile.firstName'), 'Zoe', 'belongsTo property is model has initial value');
+    assert.equal(get(dummyChangeset, 'profile.firstName'), 'Bob', 'belongsTo changeset is updated');
+    assert.equal(get(dummyChangeset.pendingData, 'profile.firstName'), 'Bob', 'belongsTo pendingData has new value');
```

Comment:
This test is marked as `todo`. In this case, when there is an error, it will be marked as "passed". When the functionality will be fixed/adapted, and it will be working - this test will fail :) I've created 2 tests specifically, one that has assert.throws another with todo, so they would fail in one or another case.

Reply:
Perfect!  I missed that.  At least we have a test for the circular issue!